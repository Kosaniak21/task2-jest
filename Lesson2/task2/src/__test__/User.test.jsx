import React from "react";
import { shallow } from "enzyme";
import User from "../User";
import { getUserData } from "../usersGateway";

jest.mock("../usersGateway", () => {
  return {
    getUserData: jest.fn(() => Promise.resolve({})),
  };
});
describe("User", () => {
  it(`should display nothing if user don't get data`, () => {
    const component = shallow(<User />);
    expect(component.find(".user").exists()).toBe(false);
  });
  it(`should display user with data`, () => {
    const component = shallow(<User />);
    component.setState({
      userDate: {
        name: "John Doe",
        location: "New York",
        avatar_url: "https://example.com",
      },
    });
    expect(component.find(".user")).toMatchSnapshot();
  });
  it(`should display nothing if user don't get data`, () => {
    shallow(<User />);

    expect(getUserData).toBeCalled();
  });
});
