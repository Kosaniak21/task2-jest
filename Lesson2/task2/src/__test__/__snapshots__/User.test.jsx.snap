// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should display user with data 1`] = `null`;
